DoLahTest/ParserUnitTest.cpp

```diff
@@ -70,6 +90,31 @@ namespace DoLahTest {
         );
     }
 
+    TEST_METHOD(parseAddDetailedTest5) {
+        parseAddTestMethod((std::string)
+            "#cs2103 #homework on the #stage from 15th to 21st",
+            { "15/" + month + "/" + year + " ~ 21/" + month + "/" + year, "{ stage, homework, cs2103 }", "#cs2103 #homework on the #stage" }
```

Comment:
Personal preference: I usually don't like hardcoded concatenation to build a string. I prefer string formatting.

This test method is quite hard to understand. I'm spending too much time trying to figure out what this line is trying to build.

Unit Test should be as simple or as trivial as possible and easy to understand. The preparation may be complex but if you abstract that, you should be able to have a much simpler unit test process.